src/Stratis.Bitcoin.Features.SmartContracts/SmartContractFeature.cs

```diff
@@ -54,7 +54,9 @@ public override Task InitializeAsync()
             if (this.network.Consensus.IsProofOfStake)
                 Guard.Assert(this.network.Consensus.ConsensusFactory is SmartContractPosConsensusFactory);
             else
-                Guard.Assert(this.network.Consensus.ConsensusFactory is SmartContractPowConsensusFactory || this.network.Consensus.ConsensusFactory is SmartContractPoAConsensusFactory);
+                Guard.Assert(this.network.Consensus.ConsensusFactory is SmartContractPowConsensusFactory
+                             || this.network.Consensus.ConsensusFactory is SmartContractPoAConsensusFactory
+                             || this.network.Consensus.ConsensusFactory is SmartContractCollateralPoAConsensusFactory);
```

Comment:
This is getting out of hand lol.... Not your fault, nothing wrong with it in this PR. 

Just shows what we need to change in future 😄

Reply:
Yeah, also thought so. 

There are many things in fedpeg that would be nice to refactor. Most importantly consensus values that are passed via commandline argument. As I've mentioned on slack it would be much nicer to have them initialized in the network and then store in some consensus storage to allow their modification via voting system in the future.

Reply:
Probably putting collateral as a factory was not the best idea, but its ok for now until Network becomes more flexible to register its dependencies.